Tools/TestWebKitAPI/Tests/WTF/darwin/MachSendRight.cpp

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (C) 2014-2022 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+#include <wtf/MachSendRight.h>
+
+#include <mach/mach.h>
+
+namespace TestWebKitAPI {
+
+static void checkSendRefs(mach_port_t port, unsigned expected)
+{
+    unsigned count;
+    auto kr = mach_port_get_refs(mach_task_self(), port, MACH_PORT_RIGHT_SEND, &count);
+    EXPECT_EQ(kr, KERN_SUCCESS);
+    EXPECT_EQ(count, expected);
+}
+
+static mach_port_t setup()
+{
+    mach_port_t port = MACH_PORT_NULL;
+    auto kr = mach_port_allocate(mach_task_self(), MACH_PORT_RIGHT_RECEIVE, &port);
+    EXPECT_EQ(kr, KERN_SUCCESS);
+
+    kr = mach_port_insert_right(mach_task_self(), port, port, MACH_MSG_TYPE_MAKE_SEND);
+    EXPECT_EQ(kr, KERN_SUCCESS);
+    
+    checkSendRefs(port, 1);
+
+    return port;
+}
+
+TEST(MachSendRight, Adopt)
+{
+    auto port = setup();
+
+    {
+        MachSendRight right = MachSendRight::adopt(port);
+        checkSendRefs(port, 1);
```

Comment:
then you could write this inline:
`EXPECT_EQ(getSendRefs(port), 1u);`

this way when it fails, it's obvious in which context it fails.